### ISSUE TITLE
FacebookLogin provider breaking changes between versions 3.2.0 - 3.5.4 and non compliant behaviour

### ISSUE THREAD

**Issue by cirix** [abacritt/angularx-social-login#355]
Hi team,
I have been using the 3.2.0 for quite sometime and I was able to get back the Social user object with a facebook object containing the data of the user (like name / url etc). 
I have update to 3.5.4 and there are some breaking changes, aka initOptions are mandatory and actually the version is mandatory. Also the FB api for login has changed and requires a few actions on the application using the library to do to be able to have access to some scopes. That is not documented clearly. What are your plans for the Facebook Login Provider?

kr
Nik

**Comment by jaibatrik** [comment]
I see. I think the code itself won't break if you pass undefined as `initOptions` (check https://github.com/abacritt/angularx-social-login/blob/07104e4c242b8e7c86b6fa110d0aea8bb0950dfc/projects/lib/src/providers/facebook-login-provider.ts), it is probably just a TS error. You can do that as a temporary solution while this gets fixed.

**Comment by cirix** [comment]
Hi @jaibatrik ,
I read the code exactly there and I passed the same init options. Looks like if no initOptions are passed the object get nulled and you get back an error from the call to FB.login stating is missing the version init. 
On the second round, you are supporting version 4, will you default to 9? Asking cause for example the response_types after version 7 are managed totally differently. Thus  4 <version<7 the code will work and SocialUser will be populated properly. Afterwards it requires manual steps when registering an application to fb developer to state what you want to grant access too.
I would document that in the lib, cause when I used the latest version, it failed miserably.

kr
\n\b
